### PR TITLE
Add manual scsi probe of a Lun

### PR DESCRIPTION
**Pull request by alessandrocarminati**
This commit introduces manual probing of a specified SCSI LUN. The LUN can be specified either via the kernel command line `scsi.addr=<host>:<channel>:<target>:<lun>` or by adding a line to the `initoverlayfs.config` file `scsi.addr <host>:<channel>:<target>:<lun>`. In case both are present, the configuration takes precedence. Any operation depends on the presence of the `scsi_mod.scan=manual` kernel argument.
If it is not present in the kernel command line, no action will occur. The directory `scsi_probe` contains the main functions needed to complete the task and some related tests.
To run the tests, simply run `make` in the `scsi_probe` subdirectory.

### CODE REVIEW

**Comment by ericcurtin** [comment]
I can't wait to give this a run!

I'm probably taking a few week off from Wednesday, so I dunno if I will get the opportunity to look at this is detail.

But it's pretty ifdef'd out, so I don't mind merging this if it helps things progress.

File: initoverlayfs.c [hunk 1]
```diff
@@ -518,7 +526,7 @@ static int switchroot(const char* newroot) {
 }
 
 static int mount_proc_sys_dev(void) {
-  if (false) {
+  if (true) {
```
[the remark this answers is not in the document]
Comment (ericcurtin) [reply initoverlayfs.c:529]: Actually, some things are broken here in general that I have to fix when I come back.

Like I think if you run the upstream version of this now on a Fedora Workstation machine, it doesn't work anymore and they are really trivial silly bugs.

I just got distracted by other things the last week or two.

**Review by ericcurtin — APPROVED** [review]

**Comment by ericcurtin** [comment]
Merging before I go on PTO, I didn't give the most thorough review, but it's all ifdef's out anyway, better to merge to not block progress...

**Comment by dougsland** [comment]
agreed, sorry i couldn't review before. If we find anything later in the road we improve it.

**Pull request merged**